src/Form/AuthenticationForm.php

```diff
@@ -637,26 +591,19 @@ protected function getActiveKey(): ?KeyInterface {
   }
 
   /**
-   * Creates a new auth key stored in a file.
+   * Creates a new auth key using the Apigee Edge Private File key provider.
    *
-   * @return \Drupal\key\KeyInterface|null
+   * @return \Drupal\key\KeyInterface
    *   A new auth key.
-   *
-   * @throws \Drupal\Core\Entity\EntityStorageException
-   * @throws \Drupal\key\Exception\KeyValueNotSetException
    */
-  protected function generateNewAuthKey(): ?KeyInterface {
-    if (!$this->fileSystem->realpath('private://')) {
-      return NULL;
-    }
-    // Create a new key name.
+  protected function generateNewAuthKey(): KeyInterface {
     $new_key_id = 'apigee_edge_connection_default';
-    $file_path = "private://.apigee_edge/{$new_key_id}.json";
-
-    // Make sure the key and the associated key file do not exist.
-    for ($i = 1; (Key::load($new_key_id) || file_exists($file_path)); $i++) {
-      $new_key_id = "apigee_edge_connection_default_{$i}";
-      $file_path = "private://.apigee_edge/{$new_key_id}.json";
+    if (Key::load($new_key_id) !== NULL) {
```

Comment:
I would inject the entity type manager - or the key entity storage - and use that to CRUDL keys because it would allow mocking CRUDL related operations in tests.